integrations/helm/status/status.go

```diff
@@ -0,0 +1,116 @@
+/*
+
+This package is for maintaining the link between `FluxHelmRelease`
+resources and the Helm releases to which they
+correspond. Specifically,
+
+ 1. updating the `FluxHelmRelease` status based on the state of the
+   associated Helm release; and,
+
+ 2. attributing each resource in a Helm release (under our control) to
+ the associated `FluxHelmRelease`.
+
+*/
+package status
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"golang.org/x/time/rate"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kube "k8s.io/client-go/kubernetes"
+	"k8s.io/helm/pkg/helm"
+
+	fluxhelmtypes "github.com/weaveworks/flux/apis/helm.integrations.flux.weave.works/v1alpha2"
+	fluxhelm "github.com/weaveworks/flux/integrations/client/clientset/versioned"
+	"github.com/weaveworks/flux/integrations/helm/release"
+)
+
+type Updater struct {
+	rateLimiter *rate.Limiter
+	fluxhelm    fluxhelm.Interface
+	kube        kube.Interface
+	helmClient  *helm.Client
+}
+
+func New(fhrClient fluxhelm.Interface, kubeClient kube.Interface, helmClient *helm.Client) *Updater {
+	limiter := rate.NewLimiter(rate.Every(10*time.Second), 5) // arbitrary
+	return &Updater{
+		rateLimiter: limiter,
+		fluxhelm:    fhrClient,
+		kube:        kubeClient,
+		helmClient:  helmClient,
+	}
+}
+
+func (a *Updater) Loop(stop <-chan struct{}, logger log.Logger) {
+	ctx := context.Background()
+	var logErr error
+
+bail:
+	for {
+		select {
+		case <-stop:
+			break bail
```

Comment:
this break is jumping before the for loop. When shutdown/stop occurs, should it be re-entering the loop?

I could be misunderstanding the use of the stop channel in this case.

Reply:
`break label` exits the block so labelled; i.e., it jumps _after_ the loop.

Reply:
ah yes, sorry